include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -2308,6 +2428,38 @@ namespace MGTransferGlobalCoarseningTools
 
 
 
+namespace
+{
+  template <int dim, typename Number>
+  void
+  weight_dofs_on_child(const VectorizedArray<Number> *weights,
+                       const unsigned int             n_components,
+                       const int                      loop_length,
+                       VectorizedArray<Number> *      data)
+  {
+    unsigned int degree_to_3[100];
+    degree_to_3[0] = 0;
+    for (int i = 1; i < loop_length - 1; ++i)
+      degree_to_3[i] = 1;
+    degree_to_3[loop_length - 1] = 2;
+    for (unsigned int c = 0; c < n_components; ++c)
+      for (int k = 0; k < (dim > 2 ? loop_length : 1); ++k)
+        for (int j = 0; j < (dim > 1 ? loop_length : 1); ++j)
+          {
+            const unsigned int shift = 9 * degree_to_3[k] + 3 * degree_to_3[j];
+            data[0] *= weights[shift];
+            // loop bound as int avoids compiler warnings in case loop_length
+            // == 1 (polynomial degree 0)
+            for (int i = 1; i < loop_length - 1; ++i)
+              data[i] *= weights[shift + 1];
+            data[loop_length - 1] *= weights[shift + 2];
+            data += loop_length;
+          }
```

Comment:
Is this the same code as in the other MG transfer function https://github.com/dealii/dealii/blob/cfa11cfe8cbc78f9a3b15bdb435fd2297d4b871d/source/multigrid/mg_transfer_matrix_free.cc#L362-L388

I suggest to move the code to a common location and only implement it once. Since this is a regular tensor product variant, it might fit in `include/deal.II/matrix_free/tensor_product_kernels.h`.